ESP8266Interface.cpp

```diff
@@ -380,11 +380,17 @@ int ESP8266Interface::socket_listen(void *handle, int backlog)
 int ESP8266Interface::socket_connect(void *handle, const SocketAddress &addr)
 {
     struct esp8266_socket *socket = (struct esp8266_socket *)handle;
+
     _esp.setTimeout(ESP8266_MISC_TIMEOUT);
 
-    const char *proto = (socket->proto == NSAPI_UDP) ? "UDP" : "TCP";
-    if (!_esp.open(proto, socket->id, addr.get_ip_address(), addr.get_port(), _local_ports[socket->id])) {
-        return NSAPI_ERROR_DEVICE_ERROR;
+    if (socket->proto == NSAPI_UDP) {
+        if (!_esp.open_udp(socket->id, addr.get_ip_address(), addr.get_port(), _local_ports[socket->id])) {
+            return NSAPI_ERROR_DEVICE_ERROR;
+        }
+    } else if (socket->proto == NSAPI_TCP) {
```

Comment:
probably just `else` here - the if is making the compiler generate code that won't do anything useful anyway.

Reply:
ok